packages/tss/src/tss.ts

```diff
@@ -0,0 +1,677 @@
+      const serverLagrangeCoeffs = selectedServerIndexes.map((x) => getLagrangeCoeffs(ec, selectedServerIndexes, x));
+      const serverInterpolated = dotProduct(serverLagrangeCoeffs, selectedServerDecs, ec.n);
+      const lagrangeCoeffs = [getLagrangeCoeffs(ec, [1, 99], 1), getLagrangeCoeffs(ec, [1, 99], 99)];
+      const tssShare = dotProduct(lagrangeCoeffs, [serverInterpolated, userDec], ec.n);
+      const tssSharePub = ec.g.mul(tssShare);
+      const tssCommitA0 = tssCommits[0];
+      const tssCommitA1 = tssCommits[1];
+      let _tssSharePub = tssCommitA0;
+      for (let j = 0; j < tssIndex; j++) {
+        _tssSharePub = _tssSharePub.add(tssCommitA1);
+      }
+      if (tssSharePub.eq(_tssSharePub)) {
+        const adjustedShare = this.adjustTssShare(tssShare, accountIndex, coefficient);
+        return { tssIndex, tssShare: adjustedShare };
+      }
+    }
+    throw new Error("could not find any combination of server decryptions that match tss commitments...");
+  }
+
+  getTSSCommits(): Point[] {
+    if (!this.privKey) throw CoreError.default("tss pub cannot be returned until you've reconstructed tkey");
+    if (!this.metadata) throw CoreError.metadataUndefined();
+    const tssPolyCommits = this.metadata.tssPolyCommits[this.tssTag];
+    if (!tssPolyCommits) throw CoreError.default(`tss poly commits not found for tssTag ${this.tssTag}`);
+    if (tssPolyCommits.length === 0) throw CoreError.default("tss poly commits is empty");
+    return tssPolyCommits;
+  }
+
+  getTSSPub(accountIndex?: number): Point {
+    if (accountIndex && this._tssKeyType === "ed25519") {
+      throw new Error("Account index not supported for ed25519");
+    }
+
+    const ec = this._tssCurve;
+    const tssCommits = this.getTSSCommits();
+    if (accountIndex && accountIndex > 0) {
+      const nonce = this.computeAccountNonce(accountIndex);
+      // we need to add the pub key nonce to the tssPub
+      const noncePub = ec.keyFromPrivate(nonce.toString("hex")).getPublic();
+      const pubKeyPoint = pointToElliptic(ec, tssCommits[0]);
+      const devicePubKeyPoint = pubKeyPoint.add(noncePub);
+      return Point.fromSEC1(devicePubKeyPoint.encodeCompressed("hex"), this.tssKeyType);
+    }
+    return tssCommits[0];
+  }
+
+  async _getRssNodeDetails(): Promise<{
+    serverEndpoints: string[];
+    serverPubKeys: PointHex[];
+    serverThreshold: number;
+  }> {
+    const { serverEndpoints, serverPubKeys, serverThreshold } = await this.serviceProvider.getRSSNodeDetails();
+    if (!Array.isArray(serverEndpoints) || serverEndpoints.length === 0) throw new Error("service provider tss server endpoints are missing");
+    if (!Array.isArray(serverPubKeys) || serverPubKeys.length === 0) throw new Error("service provider pub keys are missing");
+    return {
+      serverEndpoints,
+      serverPubKeys,
+      serverThreshold: serverThreshold || Math.floor(serverEndpoints.length / 2) + 1,
+    };
+  }
+
+  async importTssKey(
```

Comment:
do we need to handle seed for ed25519  (eg, encrypt seed with the scalar) ?
also for exportTssKey for ed25519, it is not possible to allow user to port out the seed if they use dkg generated tss key rite?

Reply:
good point.
depends how you implement it. If you import a seed, and store it to metadata, of course we can also export it. if the key is not imported, then we can't export a seed (because it would require inverting a hash function which is infeasible).

Reply:
i'll try to make a suggestion for this.

Reply:
Now supports ed25519 import seed. Does this look good? @ieow @himanshuchawla009

Reply:
this feature was there before and it works. the (pre-existing) tests also show that this was originally intended to be used to allow importing more than one key under different tags.

it's worth discussing indexed/derived accounts in the context of `ed25519` though. I am inclined to disabling this feature for the following reason: even though derived accounts will work for ed25519 tss, they will not work for single party ed25519. so users will not be able to export these accounts and use them somewhere else. (this is because ed25519 uses the seed instead of the scalar as the private key, and the seed does not have the same homomorphic properties that makes this feature possible for secp256k1.)

Reply:
after discussion: indexed accounts disabled for ed25519 for now